tools/local_provisioning_profile_finder/local_provisioning_profile_finder.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import datetime
+import plistlib
+import shutil
+import subprocess
+from typing import List, Optional, Tuple
+
+
+def _build_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("name", help="The name of the profile to find")
+    parser.add_argument("output", help="The path to copy the profile to")
+    parser.add_argument(
+        "--local-profiles",
+        nargs="*",
+        required=True,
+        help="All local provisioning profiles to search through",
+    )
+    parser.add_argument(
+        "--fallback-profiles",
+        nargs="+",
+        required=True,
+        help="Fallback provisioning profiles to use if not found locally",
+    )
+    return parser
+
+
+def _profile_contents(profile: str) -> Tuple[str, datetime.datetime]:
+    output = subprocess.check_output(["security", "cms", "-D", "-i", profile])
+    plist = plistlib.loads(output)
+    return plist["Name"], plist["CreationDate"]
+
+
+def _find_newest_profile(name: str, profiles: List[str]) -> Optional[str]:
+    sanitized_name = name.replace("_", " ")
```

Comment:
This will prevent anyone from having actual underscores in their profile names 😅

Reply:
ah yea, maybe only works for us

Reply:
Probably works for most, since I assume "spaces in the name" outscore "underscores in the name" by quite a large margin.

Worth mentioning in any docs, of course, but maybe also refer to the "design limitation" in the error message below. (Ignoring the possibility if this sanitizaition can be worked around of course – I may offer a patch later if possible. Or maybe we'll just change our profile names too)

Reply:
we can probably use something besides name instead, we could make folks pass the bundle ID since that is more stable anyways